interfaces/builtin/packagekit_control.go

```diff
@@ -0,0 +1,101 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const packageKitControlSummary = `allows control of the PackageKit service`
+
+const packageKitControlBaseDeclarationSlots = `
+  packagekit-control:
+    allow-installation:
+      slot-snap-type:
+        - core
```

Comment:
Okay, I've added an `allow-installation: false` plug declaration now, along with associated tests to show it can be overridden.

I'm still waiting on the test snap to be approved in the store though: without that the spread tests still fail.

Reply:
> I'm still waiting on the test snap to be approved in the store though: without that the spread tests still fail.

Note I approved this some time ago, but someone needs to release it to a channel.

Reply:
That's weird.  I never received an email notification from the store (I could have sworn there was one last time I went through this).

Rerunning the tests, it looks like I also need a snap-declaration assertion from the store to allow the test snap to be installed for the tests, similar to what the `test-snapd-control-consumer` snap has.

Reply:
> Rerunning the tests, it looks like I also need a snap-declaration assertion from the store to allow the test snap to be installed for the tests, similar to what the `test-snapd-control-consumer` snap has.

Done. I left it manually connected since the your tests seem to assume that.